pallets/detach/src/lib.rs

```diff
@@ -443,243 +510,256 @@ pub mod pallet {
+				storage_ref.set(&persisted_ed25119_keys);
+			}
+		}
+
+		/// Returns a Tuple of the following things:
+		/// 1. A **Signature** obtained by **signing the detach request `request` using a Fragnova-authorized account**.
 		///
-		/// Then, for each of the signed detach requests - send an unsigned transaction with a signed payload onto the Clamor Blockchain
-		/// (NOTE: the signed payload consists of a payload and a signature.
-		/// The payload is the detach request which is represented as a `DetachInternalData` struct (the struct contains the signature on the aforementioned tuple, amongst other things)
-		/// and the signature is the signature obtained from signing the aforementioned payload using `Signer::<T, T::AuthorityId>::any_account()`) (问Gio)
+		/// Note: Since it was signed by a Fragnova-authorized account, this signature can be presented to the External Blockchain `requests.target_chain`
+		/// to attach the "detached thing" (e.g a detached Proto-Fragment or a detached Fragment Instance) to the External Blockchain.
 		///
-		/// NOTE: `Signer::<T, T::AuthorityId>::any_account()` uses any of the keys that was added using the RPC `author_insertKey` into Clamor (https://polkadot.js.org/docs/substrate/rpc/#insertkeykeytype-text-suri-text-publickey-bytes-bytes)
-		fn process_detach_requests() {
+		/// 2. The Detach-Nonce of the detach request's target account. (See `DetachNonces` for more information).
+		fn get_detach_signature_and_detach_nonce(request: &DetachRequest) -> Result<(Vec<u8>, u64), Error<T>> {
+			match request.target_chain {
+				SupportedChains::EthereumMainnet
+				| SupportedChains::EthereumRinkeby
+				| SupportedChains::EthereumGoerli => {
+
+					Self::add_newly_found_ed25519_and_ecdsa_keys();
+
+					// `sp_io::crypto::ecdsa_public_keys` returns all ecdsa public keys for the given key id from the keystore. (in our case the key id is `KEY_TYPE`).
+					// Source: https://docs.rs/sp-io/latest/sp_io/crypto/fn.ecdsa_public_keys.html
+					let ecdsa_keys = Crypto::ecdsa_public_keys(KEY_TYPE);
+					log::debug!("ecdsa local keys {:x?}", ecdsa_keys);
+
+					// make sure the local key is in the global authorities set!
+					let ethereum_authority = ecdsa_keys
+						.iter()
+						.find(|k| <EthereumAuthorities<T>>::get().contains(k)).ok_or(Error::<T>::NoValidator)?;
+
+					let nonce =
+						<DetachNonces<T>>::get(&request.target_account, request.target_chain).unwrap_or_default().checked_add(1).unwrap();
+					let mut chain_id_be: [u8; 32] = [0u8; 32]; // "be" stands for big-endian
+					match request.target_chain {
+						SupportedChains::EthereumMainnet => U256::from(1),
+						SupportedChains::EthereumRinkeby => U256::from(4),
+						SupportedChains::EthereumGoerli => U256::from(5),
+					}.to_big_endian(&mut chain_id_be);
+
+					let payload = [
+						// REVIEW - `request.hash` is a `DetachHash` enum. Do we want that? @sinkingsugar
+						&request.hash.encode()[..],
```

Comment:
The payload's (which will then signed) first piece of data is `request.hash.encode()`, where `request.hash` is a `DetachHash` enum. 

Do we want that? @sinkingsugar 

Because if we do it like this, we would have to use SCALE encoding in Fragnova's Ethereum Smart Contract also.

Reply:
Actually it would be annoying since we would have to decipher the SCALE encoded Rust enum to either a Proto-Fragment or a Fragment Instance in our EVM Smart Contract to understand if it's a Proto or a FI!

I think it would be better to just add the Proto-Fragment's Hash or the Fragment Instance's Tuple into the payload then right? and we can just concat the string "proto" to the payload if we wanna detatch/attach a Proto-Fragment or concat the string "instance" to the payload if we wanna detach/attach a Fragment Instance?

Reply:
Personally, I would avoid to have to decipher the SCALE encoded Rust enum. At the moment our smart contract receives the hash, I would keep it simple.

Reply:
> Actually it would be annoying since we would have to decipher the SCALE encoded Rust enum to either a Proto-Fragment or a Fragment Instance in our EVM Smart Contract to understand if it's a Proto or a FI!
> 
> I think it would be better to just add the Proto-Fragment's Hash or the Fragment Instance's Tuple into the payload then right? and we can just concat the string "proto" to the payload if we wanna detatch/attach a Proto-Fragment or concat the string "instance" to the payload if we wanna detach/attach a Fragment Instance?

so this is a good solution, right? @alebaffa @sinkingsugar ?

Reply:
Okay I've simplified it (just pushed) @alebaffa